src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -184,4 +189,6 @@ object BlockExecutionError {
   case object MissingParentError extends BlockExecutionError {
     override val reason: Any = "Cannot find parent"
   }
+
+  case class MPTError(reason: Throwable) extends BlockExecutionError
```

Comment:
We should make all new case classes `final` (and also existing ones that you come across)...this will be enforced by code linting soon hopefully.